lib/manageiq/api/client/connection.rb

```diff
@@ -49,6 +49,12 @@ def json_response
           raise JSON::ParserError, "Response received from #{url} is not of type #{CONTENT_TYPE}"
         end
 
+        def fetch_token
+          @authentication.token = nil
+          res = get "auth"
+          @authentication.token = res["auth_token"]
+        end
```

Comment:
I take it this is a helper method so do an auth.  Would prefer do have an auth method on the client instance (miq), delegated to here is fine. i.e. miq.auth, returning and AuthResponse object, but have the caller specifically call an update_authentication with the new token,  miq.update_authentication could be called with user, password, token or group.